File: tests/js/vertexai/test_openllmetry.ts
/**
 * Conformance test: OpenLLMetry (Traceloop) JS Vertex AI instrumentation.
 */

import { VertexAIInstrumentation } from "@traceloop/instrumentation-vertexai";
import { run } from "./common";

function instrument(vertexaiModule: any) {
  const instrumentation = new VertexAIInstrumentation();
  instrumentation.manuallyInstrument(vertexaiModule);
}

run("OpenLLMetry JS: Vertex AI Conformance Test", instrument).catch((e) => { console.error(e); process.exit(1); });

File: tests/python/groq/test_openllmetry.py
"""Conformance test: OpenLLMetry (Traceloop) Groq instrumentation."""

from common import run, run_chat, run_chat_streaming


def instrument():
    from opentelemetry.instrumentation.groq import GroqInstrumentor
    GroqInstrumentor().instrument()


if __name__ == "__main__":
    run(
        "OpenLLMetry: Groq Conformance Test",
        instrument,
        [run_chat, run_chat_streaming],
    )
